packages/@expo/cli/src/run/android/index.ts

```diff
@@ -47,6 +48,7 @@ export const expoRunAndroid: Command = async (argv) => {
     --variant <name>       Build variant. {dim Default: debug}
     -d, --device [device]  Device name to run the app on
     -p, --port <port>      Port to start the dev server on. {dim Default: 8081}
+    --active-arch-only     Only build for the architectures of connected devices or emulators
```

Comment:
I think we should probably turn this feature on by default and possibly add an environment variable to disable setting `PreactNativeArchitectures` or configure the exact value.

Reply:
I also agree it should be on by default. And yes this feature is meant to be used only for dev builds.